lib/Doctrine/DBAL/Platforms/SQLAnywherePlatform.php

```diff
@@ -1310,16 +1310,14 @@ protected function doModifyLimitQuery($query, $limit, $offset)
     {
         $limitOffsetClause = '';
 
-        if ($limit > 0) {
+        if ($limit !== null) {
             $limitOffsetClause = 'TOP ' . $limit . ' ';
-        }
 
-        if ($offset > 0) {
-            if ($limit === 0) {
-                $limitOffsetClause = 'TOP ALL ';
+            if ($offset > 0) {
+                $limitOffsetClause .= 'START AT ' . ($offset + 1) . ' ';
```

Comment:
Can you move the code that generates `$limitOffsetClause` to a `private` method? All of these if/else become early returns that way.